src/tools/w3dviewer/camerasettingsdialog.cpp

```diff
@@ -0,0 +1,229 @@
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "camerasettingsdialog.h"
+#include "camera.h"
+#include "graphicview.h"
+#include "resource.h"
+#include "utils.h"
+#include "viewerscene.h"
+#include "w3dviewdoc.h"
+
+bool CameraSettingsDialogClass::m_isUpdating = false;
+
+// clang-format off
+BEGIN_MESSAGE_MAP(CameraSettingsDialogClass, CDialog)
+	ON_COMMAND(IDC_FOV, OnFieldOfView)
+	ON_COMMAND(IDC_CLIPPLANES, OnClipPlanes)
+	ON_COMMAND(IDABORT, OnReset)
+END_MESSAGE_MAP()
+// clang-format on
+
+CameraSettingsDialogClass::CameraSettingsDialogClass(CWnd *pParentWnd) : CDialog(IDD_CAMERA, pParentWnd) {}
+
+void CameraSettingsDialogClass::DoDataExchange(CDataExchange *pDX)
+{
+    DDX_Control(pDX, IDC_LENSFOVSPIN, m_lensSpin);
+    DDX_Control(pDX, IDC_FARCLIPSPIN, m_farClipSpin);
+    DDX_Control(pDX, IDC_VERTFOVSPIN, m_vertFovSpin);
+    DDX_Control(pDX, IDC_NEARCLIPSPIN, m_nearClipSpin);
+    DDX_Control(pDX, IDC_HORFOVSPIN, m_horizFovSpin);
+}
+
+BOOL CameraSettingsDialogClass::OnCommand(WPARAM wParam, LPARAM lParam)
+{
+    if (!m_isUpdating) {
+        if (LOWORD(wParam) == IDC_HORFOV) {
+            if (HIWORD(wParam) == EN_UPDATE) {
+                m_isUpdating = true;
+                double d = GetDlgItemFloat(m_hWnd, IDC_HORFOV);
+
+                if (d > 0.0) {
+                    float f1 = 0.01799999922513962f / tan(d * 0.01745329300562541 * 0.5) * 1000.0f;
+                    SetDlgItemFloat(m_hWnd, IDC_LENSFOV, f1);
+                }
+
+                m_isUpdating = false;
+            }
+        } else if ((LOWORD(wParam) == IDC_LENSFOV) && (HIWORD(wParam) == EN_UPDATE)) {
+            m_isUpdating = true;
+            double d = GetDlgItemFloat(m_hWnd, IDC_LENSFOV) * 0.001f;
+
+            if (d > 0.0) {
+                float f1 = atan2(0.017999999 / d, 1.0) + atan2(0.017999999 / d, 1.0);
+                float f2 = f1 * 57.29577791868204f;
+                SetDlgItemFloat(m_hWnd, IDC_HORFOV, f2);
+                float f3 = f1 * 0.75f * 57.29577791868204f;
+                SetDlgItemFloat(m_hWnd, IDC_VERTFOV, f3);
+            }
+
+            m_isUpdating = false;
+        }
+    }
+
+    return CWnd::OnCommand(wParam, lParam);
+}
+
+BOOL CameraSettingsDialogClass::OnNotify(WPARAM wParam, LPARAM lParam, LRESULT *pResult)
+{
+    NMUPDOWN *ud = reinterpret_cast<NMUPDOWN *>(lParam);
+
+    if (ud != nullptr && ud->hdr.code == UDN_DELTAPOS) {
+        UpdateEditCtrl(ud->hdr.hwndFrom, ud->iDelta);
+
+        if (ud->hdr.idFrom == IDC_LENSFOVSPIN) {
+            double d = GetDlgItemFloat(m_hWnd, IDC_LENSFOV) * 0.001f;
+
+            if (d > 0.0) {
+                float f1 = atan2(0.017999999 / d, 1.0) + atan2(0.017999999 / d, 1.0);
```

Comment:
0.018 perhaps?

In `CameraSettingsDialogClass::OnReset()` function 0.018 was used.